quickwit-search/src/collector.rs

```diff
@@ -280,16 +280,18 @@ impl Collector for QuickwitCollector {
         // starting from 0 for every leaves.
         let leaf_max_hits = self.max_hits + self.start_offset;
 
-        let timestamp_filter_opt = if let Some(timestamp_field) = self.timestamp_field_opt {
-            TimestampFilter::new(
+        // Create a timestamp filter only if necessary as, once created,
+        // we will read some bytes in the fast field file that are not in the hotcache.
+        let timestamp_filter_opt = self.timestamp_field_opt
+            .filter(|_| self.start_timestamp_opt.is_some() || self.end_timestamp_opt.is_some())
+            .map(|timestamp_field| TimestampFilter::new(
```

Comment:
We should use split metadata to create a timestamp filter only if there is an overlap between the query filter and the split start / end timestamp.

Reply:
I don't understand...
Split pruning already happened at this point?

Reply:
I need to think a bit more about this because this is not very readable.
But currently, what I want to avoid is to create a `TimestampFilter` when there is no timestamp pruning because instanciating a `TimestampFilter` will lead a read of fast field min and max.

Reply:
> We should use split metadata to create a timestamp filter only if there is an overlap between the query filter and the split start / end timestamp.

I think I understand what you meant but as far as I can tell, your formulation is wrong.

- No overlap: the leaf split search will not be emitted anyway, there is no problem today
- Split timestamps entirely included in the search timestamp range: We do not need ot warmup the timestamp field and we do not need the filter. Right now we forget the timestamp filter. I opened #1748 accordingly.
- Other (= some overlap but not included)